types/index.d.ts

```diff
@@ -2697,7 +2699,86 @@ declare namespace math {
     to(x: Unit | MathArray | Matrix, unit: Unit | string): Unit | MathArray | Matrix;
 
     /*************************************************************************
-     * Utils functions
+     * Utils
+     ************************************************************************/
+     isNumber(x: unknown): x is number;
+
+     isBigNumber(x: unknown): x is BigNumber;
+
+     isComplex(x: unknown): x is Complex;
+
+     isFraction(x: unknown): x is Fraction;
+
+     isUnit(x: unknown): x is Unit;
+
+     isString(x: unknown): x is string;
+
+     isArray: ArrayConstructor['isArray'];
+
+     isMatrix(x: unknown): x is Matrix;
+
+     isCollection(x: unknown): x is (Matrix | any[]);
+
+     isDenseMatrix(x: unknown): x is Matrix;
+
+     isSparseMatrix(x: unknown): x is Matrix;
+
+     isRange(x: unknown): boolean;
+
+     isIndex(x: unknown): x is Index;
+
+     isBoolean(x: unknown): x is boolean;
+
+     isResultSet(x: unknown): boolean;
+
```

Comment:
This is a similar situation. A ResultSet can be returned from a call to `math.evaluate()`, which at the moment has the less-than-fully specific TypeScript type `any`. In the same spirit, ResultSet should be added to index.d.ts, and the return type of evaluate should be improved, and this should be changed to a guard. So again: in or out of scope for this PR?

Reply:
I'd prefer to add that in a separate PR, which I'm happy to do. The main reason I didn't include it was because it didn't a type declared and I wasn't particularly familiar with it. I've made isChain a type predicate since that already has a type defined.